flytestdlib/cache/auto_refresh.go

```diff
@@ -275,17 +295,34 @@ func (w *autoRefresh) sync(ctx context.Context) (err error) {
 		default:
 			item, shutdown := w.workqueue.Get()
 			if shutdown {
+				logger.Debugf(ctx, "Shutting down worker")
 				return nil
 			}
 
-			t := w.metrics.SyncLatency.Start()
-			updatedBatch, err := w.syncCb(ctx, *item.(*Batch))
-
 			// Since we create batches every time we sync, we will just remove the item from the queue here
 			// regardless of whether it succeeded the sync or not.
 			w.workqueue.Forget(item)
 			w.workqueue.Done(item)
 
+			t := w.metrics.SyncLatency.Start()
+			batch := *item.(*Batch)
+			if len(batch) == 1 {
```

Comment:
I think this block should instead do something like:
```
newBatch := make(Batch, 0, len(batch))
for _, itemID := range batch {
    item, ok := w.lruMap.Get(itemID)
    if !ok {
     ...
     continue
    }

    if is terminal {
    ..... continue
    }

    b = append(b, item)
}

syncB(...b)

Reply:
done